### ISSUE TITLE
Consider supporting bounds for `NonlinearProblem` resolution

### ISSUE THREAD

**Issue by BambOoxX** [SciML/NonlinearSolve.jl#183]
This issue is based on the discussion in [here](https://discourse.julialang.org/t/how-does-modelingtoolkit-use-bounds-metadata/81487/8)

The idea would be to support resolution of systems of nonlinear equations subject to contraints such as the ones available with JuMP https://jump.dev/JuMP.jl/stable/installation/#Supported-solvers

**Comment by ValentinKaisermayer** [comment]
Isn't that just an OptimizationProblem with a dummy objective function, i.e. a feasibility problem.

**Comment by ChrisRackauckas** [comment]
Yes, feasibility problems are nonlinear problems. We should have such conversions automated in the near future, that would be nice. But one of the things prohibiting that right now is indeed the lack of bounds support in NonlinearSolve.jl

**Comment by ChrisRackauckas** [comment]
Bounds are now supported.
